src/Illuminate/Database/Connection.php

```diff
@@ -817,6 +823,17 @@ public function logQuery($query, $bindings, $time = null)
         }
     }
 
+    /**
+     * Detects if the database exception was caused by a unique constraint violation.
+     *
+     * @return bool
+     */
+    protected function matchesUniqueConstraintException(Exception $exception)
+    {
+        // It's up to each child class of Connection to detect a unique constraint violation.
+        return false;
```

Comment:
Not sure about this default implementation. Please, let me know what y'all think about this. The only issue I see here is that custom drivers should implement this method, otherwise the `createOrFirst` won't work.